packages/forklift-console-plugin/src/modules/Overview/views/overview/tabs/Details/cards/ChartsCard.tsx

```diff
@@ -0,0 +1,196 @@
+  CardHeader,
+  CardTitle,
+  Dropdown,
+  DropdownItem,
+  Flex,
+  KebabToggle,
+} from '@patternfly/react-core';
+import chart_color_blue_200 from '@patternfly/react-tokens/dist/esm/chart_color_blue_200';
+import chart_color_green_400 from '@patternfly/react-tokens/dist/esm/chart_color_green_400';
+import chart_color_red_100 from '@patternfly/react-tokens/dist/esm/chart_color_red_100';
+
+import { MigrationsCardProps } from './MigrationsCard';
+
+interface MigrationDataPoint {
+  dateLabel: string;
+  value: number;
+}
+
+const toStarted = (m: V1beta1Migration): string => m.status.started;
+const toFinished = (m: V1beta1Migration): string => m.status.completed;
+const hasTimestamp = (timestamp: string) => timestamp && DateTime.fromISO(timestamp).isValid;
+const toDateTime = (timestamp: string): DateTime => DateTime.fromISO(timestamp);
+const isLast7Days = (date: DateTime) => date.diffNow('days').get('days') <= 7;
+const isLast24H = (date: DateTime) => date.diffNow('hours').get('hours') <= 24;
+const toDayLabel = (date: DateTime): string => date.toFormat('LLL dd');
+const toHourLabel = (date: DateTime): string => date.toFormat('HH:mm');
+const createTimeBuckets = (isDaysViewSelected: boolean) =>
+  Interval.fromDateTimes(
+    DateTime.now()
+      .minus(isDaysViewSelected ? { days: 7 } : { hours: 24 })
+      // adjust the time window granularity i.e.
+      // assume 24h window and current time 14:30
+      // event that happened at 14:10 on the previous day is older then 24h when calculated with minute-precision
+      // but should be included with hour-precision (as we show on the chart)
+      .startOf(isDaysViewSelected ? 'day' : 'hour'),
+    DateTime.now().endOf(isDaysViewSelected ? 'day' : 'hour'),
+  )
+    .splitBy(isDaysViewSelected ? { day: 1 } : { hour: 4 })
+    .map((interval) => [interval, []]);
+
+const groupByBucket = (acc: [Interval, DateTime[]][], date: DateTime) =>
+  acc.map(([interval, points]) =>
+    interval.contains(date) ? [interval, [...points, date]] : [interval, points],
+  );
+
+const toDataPoints = (
+  allMigrations: V1beta1Migration[],
+  toTimestamp: (m: V1beta1Migration) => string,
+  isDaysViewSelected: boolean,
+): MigrationDataPoint[] =>
+  allMigrations
+    .map(toTimestamp)
+    .filter(hasTimestamp)
+    .map(toDateTime)
+    .filter(isDaysViewSelected ? isLast7Days : isLast24H)
+    .reduce(groupByBucket, createTimeBuckets(isDaysViewSelected))
+    .map(([interval, points]) => ({
+      dateLabel: isDaysViewSelected ? toDayLabel(interval.start) : toHourLabel(interval),
+      value: points.length,
+    }));
+
+export const ChartsCard: React.FC<MigrationsCardProps> = () => {
+  const { t } = useForkliftTranslation();
+  const [isDropdownOpened, setIsDropdownOpened] = useState(false);
+  const onToggle = () => setIsDropdownOpened(!isDropdownOpened);
+  const [isDaysViewSelected, setIsDaysViewSelected] = useState(true);
+  const [allMigrations] = useK8sWatchResource<V1beta1Migration[]>({
```

Comment:
```suggestion
  const [migrations] = useK8sWatchResource<V1beta1Migration[]>({
```

the data for the charts will need different name